diff -up e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update.group-attrs e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update
diff -up e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin.group-attrs e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin
diff -up e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update.group-attrs e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update
--- e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update.group-attrs	2010-11-04 23:03:07.000000000 -0600
+++ e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update	2010-11-04 23:05:21.000000000 -0600
@@ -207,8 +207,7 @@ foreach my $group (qw/nobody shared www/
     $updates->{$dn}->{objectClass} = ['posixGroup'];
 
     # Read information from getent group
-    @{$updates->{$dn}}{'cn','userPassword','gidNumber','memberUid'} = getgrnam($group);
-    $updates->{$dn}->{userPassword} =~ s/^/{CRYPT}/ unless $updates->{$dn}->{userPassword} =~ m/^{/;
+    @{$updates->{$dn}}{'cn','junk','gidNumber','memberUid'} = getgrnam($group);
     $updates->{$dn}->{memberUid} = [ split /\s+/, $updates->{$dn}->{memberUid} ];
 
     # Ensure groups have the needed properties
@@ -311,9 +310,7 @@ foreach my $acct (@accounts)
 	$updates->{$dn}->{objectClass} = ['posixGroup'];
 
 	# Read information from getent group
-	@{$updates->{$dn}}{'cn','userPassword','gidNumber','memberUid'} = getgrnam($key);
-	$updates->{$dn}->{userPassword} =~ s/^/{CRYPT}/ unless $updates->{$dn}->{userPassword} =~ m/^{/;
-	$updates->{$dn}->{description} = $desc if $desc;
+	@{$updates->{$dn}}{'cn','junk','gidNumber','memberUid'} = getgrnam($key);
 	$updates->{$dn}->{memberUid} = [ split /\s+/, $updates->{$dn}->{memberUid} ];
 
 	# Ensure groups have the needed properties
@@ -321,6 +318,7 @@ foreach my $acct (@accounts)
 	{
 	    push @{$updates->{$dn}->{objectClass}}, 'mailboxRelatedObject';
 	    $updates->{$dn}->{mail} = "$key\@$domain";
+	    $updates->{$dn}->{description} = $desc if $desc;
 	}
 
 	# Samba parameters if we find the samba.schema
diff -up e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin.group-attrs e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin
--- e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin.group-attrs	2010-11-04 23:03:07.000000000 -0600
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin	2010-11-04 23:04:08.000000000 -0600
@@ -97,7 +97,7 @@
     endpwent();
 
     # Read in group details
-    while(my ($key,$pwd,$gid,$members) = getgrent())
+    while(my ($key,undef,$gid,$members) = getgrent())
     {
         # skip non sme groups
         $account = $a->get($key) || next;
@@ -111,15 +111,14 @@
 
         $ldif->{$dn}->{objectClass} = [ qw(posixGroup) ];
         $ldif->{$dn}->{cn} = $key;
-        $ldif->{$dn}->{userPassword} = ($pwd =~ m/^\{/) ? $pwd : "\{CRYPT\}$pwd";
         $ldif->{$dn}->{gidNumber} = $gid;
-        $ldif->{$dn}->{description} = $ldif->{$dnmap->{$key}->{user}}->{cn} if exists $ldif->{$dnmap->{$key}->{user}}->{cn};
         $ldif->{$dn}->{memberUid} = [ split /\s+/, $members ];
 
         if ($type eq 'group')
         {
             push @{$ldif->{$dn}->{objectClass}}, 'mailboxRelatedObject';
             $ldif->{$dn}->{mail} = "$key\@$DomainName";
+            $ldif->{$dn}->{description} = $ldif->{$dnmap->{$key}->{user}}->{cn} if exists $ldif->{$dnmap->{$key}->{user}}->{cn};
         }
     }
     endgrent();
@@ -133,9 +132,7 @@
         $dnmap->{$group}->{group} = $dn;
 
         $ldif->{$dn}->{objectClass} = [ qw(posixGroup) ];
-        @{$ldif->{$dn}}{'cn','userPassword','gidNumber','memberUid'} = getgrnam($group);
-        $ldif->{$dn}->{userPassword} =~ s/^/{CRYPT}/ unless $ldif->{$dn}->{userPassword} =~ m/^\{/;
-        $ldif->{$dn}->{description} = $ldif->{$dnmap->{$key}->{user}}->{cn} if exists $ldif->{$dnmap->{$key}->{user}}->{cn};
+        @{$ldif->{$dn}}{'cn','junk','gidNumber','memberUid'} = getgrnam($group);
         $ldif->{$dn}->{memberUid} = [ split /\s+/, $ldif->{$dn}->{memberUid} ];
 
         if ($group eq 'shared'){
