diff -Naur e-smith-ldap-5.2.0.orig/root/etc/e-smith/events/actions/ldap-update e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update
--- e-smith-ldap-5.2.0.orig/root/etc/e-smith/events/actions/ldap-update	2010-04-30 17:18:31.596869272 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update	2010-04-30 17:20:06.764160119 +0200
@@ -67,6 +67,8 @@
     die "Account $userName not found.\n" unless defined $account;
     my $type = $account->prop('type') || "unknown";
 
+    exit (0) if ($type eq 'ibay');
+
     die "Account $userName is not a user or group account; " .
 	    "update LDAP entry failed.\n"
 	unless (($type eq 'user') || ($type eq 'group') || ($userName eq 'admin'));
