diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-5.2.0/root/etc/e-smith/db/configuration/force/ldap/status mezzanine_patched_e-smith-ldap-5.2.0/root/etc/e-smith/db/configuration/force/ldap/status
--- e-smith-ldap-5.2.0/root/etc/e-smith/db/configuration/force/ldap/status	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-ldap-5.2.0/root/etc/e-smith/db/configuration/force/ldap/status	2010-09-09 15:05:14.000000000 +0200
@@ -0,0 +1 @@
+enabled
