--- e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory.add2general2	2008-07-04 23:13:37.000000000 +0200
+++ e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory	2008-07-04 23:13:37.000000000 +0200
@@ -106,7 +106,7 @@
             type="select"
             id="Access"
             value="get_prop('ldap','access')"
-            options="'private' => 'PRIVATE', 'public'  => 'PUBLIC'">
+            options="'private' => 'NETWORKS_ALLOW_LOCAL', 'public'  => 'NETWORKS_ALLOW_PUBLIC'">
             <label>DIRECTORY_ACCESS</label>
             <description>DESC_DIRECTORY_ACCESS</description>
         </field>
--- e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory.add2general3	2008-07-04 23:22:42.000000000 +0200
+++ e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory	2008-07-04 23:24:11.000000000 +0200
@@ -3,14 +3,6 @@
      <base>FORM_TITLE</base>
        <trans>Change LDAP directory settings</trans>
    </entry>	
-   <entry>
-     <base>PUBLIC</base>
-      <trans>Allow public access (entire Internet)</trans>
-   </entry>
-    <entry>
-     <base>PRIVATE</base>
-      <trans>Allow access only from local networks</trans>
-  </entry>
 <entry>
 <base>LABEL_ROOT</base>
 <trans>Server root</trans>
