diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/events/actions/ldap-update e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/events/actions/ldap-update	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update	2010-01-13 16:38:40.000000000 +0100
@@ -118,10 +118,7 @@
 	utf8::upgrade($city);
 	my $street = $acct->prop('Street') || '';
 	utf8::upgrade($street);
-	my $freebusy = $acct->prop('FreeBusy') || 
-	    'https://'.$hostname.'.'.$domain.'/horde/kronolith/fb.php?u='.$key.'@'.$domain;
-	utf8::upgrade($freebusy);
-	push @attrs, (objectClass => 'person');
+	push @attrs, (objectClass => 'inetOrgPerson');
 	push @attrs, (uid => $key);
 
 	push @attrs, (cn => $name) unless ($name =~ /^\s*$/);
@@ -133,12 +130,12 @@
 	push @attrs, (ou => $dept) unless $dept =~ /^\s*$/;
 	push @attrs, (l => $city) unless $city =~ /^\s*$/;
 	push @attrs, (street => $street) unless $street =~ /^\s*$/;
-	push @attrs, (calFBURL => $freebusy) unless $freebusy =~ /^\s*$/;
     }
     elsif ($type eq 'group')
     {
-	push @attrs, (objectClass => 'posixGroup');
+	push @attrs, (objectClass => 'inetOrgPerson');
 	push @attrs, (uid => $key);
+	push @attrs, (sn => $key);
 
 	my $key = $acct->key;
 	my $desc = $acct->prop('Description') || '';
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/etc/openldap/slapd.conf/11rfc2739schema e-smith-ldap-5.2.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/11rfc2739schema
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/etc/openldap/slapd.conf/11rfc2739schema	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/11rfc2739schema	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-include         /etc/openldap/schema/rfc2739.schema
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	2010-01-13 16:38:40.000000000 +0100
@@ -1,4 +1,10 @@
 {
+    my ($dc,undef) = split (/\./, $DomainName);
+    my $o = $ldap{defaultCompany};
+
     $OUT .= "dn: $ldapBase\n";
     $OUT .= "objectClass: organization\n";
+    $OUT .= "dc: $dc\n";
+    $OUT .= "o: $o\n";
+    $OUT .= "objectClass: dcObject\n";
 }
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2006-03-15 22:17:43.000000000 +0100
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2010-01-13 16:38:40.000000000 +0100
@@ -6,8 +6,10 @@
 
 	$OUT .= "\n";
 	$OUT .= "dn: uid=$key,$ldapBase\n";
-	$OUT .= "objectClass: group\n";
+	$OUT .= "objectClass: inetOrgPerson\n";
 	$OUT .= "mail: $key\@$DomainName\n";
 	$OUT .= utf8("cn: $desc\n") if $desc;
+	$OUT .= "uid: $key\n";
+	$OUT .= "sn: $key\n";
     }
 }
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2005-04-15 21:30:38.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2010-01-13 16:38:40.000000000 +0100
@@ -13,7 +13,7 @@
 
 	$OUT .= "\n";
 	$OUT .= utf8("dn: uid=$key,$ldapBase\n");
-	$OUT .= utf8("objectClass: person\n");
+	$OUT .= utf8("objectClass: inetOrgPerson\n");
 	$OUT .= utf8("uid: $key\n");
 	$OUT .= utf8("cn: $name\n") if $name;
 	$OUT .= utf8("givenName: $first\n") if $first;
diff -Naur e-smith-ldap-5.2.0-old/root/etc/openldap/schema/rfc2739.schema e-smith-ldap-5.2.0/root/etc/openldap/schema/rfc2739.schema
--- e-smith-ldap-5.2.0-old/root/etc/openldap/schema/rfc2739.schema	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/openldap/schema/rfc2739.schema	1970-01-01 01:00:00.000000000 +0100
@@ -1,98 +0,0 @@
-#
-# http://www.faqs.org/rfcs/rfc2739.html
-# 
-# From the RFC:
-# The calCalURI contains the URI to a snapshot of the user's entire
-# default calendar. The calFBURL contains the URI to the user's default
-# busy time data. The calCAPURI represents contains a URI that can be
-# used to communicate with the user's calendar. The calCalAdrURI
-# contains a URI that points to the location to which event requests
-# should be sent for that user.
-#
-# The calOtherCalURIs is a multi-valued property containing URIs to
-# snapshots of other calendars that the user may have. The
-# calOtherFBURLs is a multi-valued property containing URIs to other
-# free/busy data that the user may have. The calOtherCAPURIs attribute
-# is a multi-valued property containing URIs to other calendars that
-# the user may have. The calOtherCalAdrURIs attribute is a multi-valued
-# property containing URIs to other locations that a user may want
-# event requests sent to.
-#
-# There is no predetermined order to the values in either multi-valued
-# property.
-
-#	EQUALITY caseIgnoreIA5Match
-
-attribute (1.2.840.113556.1.4.478
-  NAME 'calCalURI'
-  DESC 'Snapshot of users entire default calendar'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.479
-  NAME 'calFBURL'
-  DESC 'URI of the uses free and busy information'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.480
-  NAME 'calCAPURI'
-  DESC 'URI used to communicate with the users calendar'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.481
-  NAME 'calCalAdrURI'
-  DESC 'URI to which event requests should be sent for the user'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.482
-  NAME 'calOtherCalURIs'
-  DESC 'URIs to non-default calendars belonging to the user'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.483
-  NAME 'calOtherFBURLs'
-  DESC 'URIs to non-default free and busy information files'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.484
-  NAME 'calOtherCAPURIs'
-  DESC 'URIs for communicating with non-default calendars'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-attribute (1.2.840.113556.1.4.485
-  NAME 'calOtherCalAdrURIs'
-  DESC 'Destinations for event requests to non-default calendars'
-  EQUALITY caseIgnoreIA5Match
-  SUBSTR caseIgnoreIA5SubstringsMatch
-  SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
- )
-
-objectclass (1.2.840.113556.1.5.87
- NAME 'calEntry' 
- DESC 'Calendering and Free Busy information'
- SUP top AUXILIARY
- MAY (calCalURI $ calFBURL $ calCAPURI $ calCalAdrURI $ 
-      calOtherCalURIs $ calOtherFBURLs $ calOtherCAPURIs $
-      calOtherCalAdrURIs
-     )
- )
