diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/25pidfile mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/25pidfile
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/25pidfile	2002-03-25 14:17:42.000000000 -0500
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/25pidfile	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-pidfile		/var/run/slapd.pid
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/30argsfile mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/30argsfile
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/30argsfile	2002-03-25 14:17:42.000000000 -0500
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/30argsfile	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-argsfile	/var/run/slapd.args
