diff -up e-smith-ldap-5.2.0/createlinks.ldap-update-link-in-user-lock-event e-smith-ldap-5.2.0/createlinks
--- e-smith-ldap-5.2.0/createlinks.ldap-update-link-in-user-lock-event	2010-01-30 20:09:17.000000000 +0100
+++ e-smith-ldap-5.2.0/createlinks	2010-01-30 20:11:05.000000000 +0100
@@ -17,6 +17,7 @@ event_link("ldap-update", "group-create"
 event_link("ldap-delete", "group-delete", "25");
 event_link("ldap-update", "user-create", "25");
 event_link("ldap-delete", "user-delete", "25");
+event_link("ldap-update", "user-lock", "25");
 
 event_link("ldap-update", "user-modify", "25");
 event_link("ldap-update", "group-modify", "25");
