diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-5.2.0/createlinks mezzanine_patched_e-smith-ldap-5.2.0/createlinks
--- e-smith-ldap-5.2.0/createlinks	2010-10-11 20:39:52.000000000 +0200
+++ mezzanine_patched_e-smith-ldap-5.2.0/createlinks	2010-10-11 20:39:32.000000000 +0200
@@ -25,6 +25,12 @@
 event_link("ldap-update", "group-modify", "25");
 event_link("ldap-update", "password-modify", "25");
 
+event_link("ldap-update", "ibay-create", "25");
+event_link("ldap-update", "ibay-modify", "25");
+event_link("ldap-delete", "ibay-delete", "25");
+
+event_link("ldap-update", "machine-account-create", "25");
+
 event_link("ldap-dump", "pre-backup", "30");
 
 event_link("ldap-update",   "ldap-update", "80");
