diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/events/actions/ldap-update e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/events/actions/ldap-update	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/events/actions/ldap-update	2010-01-13 16:38:40.000000000 +0100
@@ -118,10 +118,7 @@
 	utf8::upgrade($city);
 	my $street = $acct->prop('Street') || '';
 	utf8::upgrade($street);
-	my $freebusy = $acct->prop('FreeBusy') || 
-	    'https://'.$hostname.'.'.$domain.'/horde/kronolith/fb.php?u='.$key.'@'.$domain;
-	utf8::upgrade($freebusy);
-	push @attrs, (objectClass => 'person');
+	push @attrs, (objectClass => 'inetOrgPerson');
 	push @attrs, (uid => $key);
 
 	push @attrs, (cn => $name) unless ($name =~ /^\s*$/);
@@ -133,12 +130,12 @@
 	push @attrs, (ou => $dept) unless $dept =~ /^\s*$/;
 	push @attrs, (l => $city) unless $city =~ /^\s*$/;
 	push @attrs, (street => $street) unless $street =~ /^\s*$/;
-	push @attrs, (calFBURL => $freebusy) unless $freebusy =~ /^\s*$/;
     }
     elsif ($type eq 'group')
     {
-	push @attrs, (objectClass => 'posixGroup');
+	push @attrs, (objectClass => 'inetOrgPerson');
 	push @attrs, (uid => $key);
+	push @attrs, (sn => $key);
 
 	my $key = $acct->key;
 	my $desc = $acct->prop('Description') || '';
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	2008-08-20 23:26:21.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	2010-01-13 16:38:40.000000000 +0100
@@ -1,4 +1,10 @@
 {
+    my ($dc,undef) = split (/\./, $DomainName);
+    my $o = $ldap{defaultCompany};
+
     $OUT .= "dn: $ldapBase\n";
     $OUT .= "objectClass: organization\n";
+    $OUT .= "dc: $dc\n";
+    $OUT .= "o: $o\n";
+    $OUT .= "objectClass: dcObject\n";
 }
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2006-03-15 22:17:43.000000000 +0100
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2010-01-13 16:38:40.000000000 +0100
@@ -6,8 +6,10 @@
 
 	$OUT .= "\n";
 	$OUT .= "dn: uid=$key,$ldapBase\n";
-	$OUT .= "objectClass: group\n";
+	$OUT .= "objectClass: inetOrgPerson\n";
 	$OUT .= "mail: $key\@$DomainName\n";
 	$OUT .= utf8("cn: $desc\n") if $desc;
+	$OUT .= "uid: $key\n";
+	$OUT .= "sn: $key\n";
     }
 }
diff -Naur e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users
--- e-smith-ldap-5.2.0-old/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2005-04-15 21:30:38.000000000 +0200
+++ e-smith-ldap-5.2.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2010-01-13 16:38:40.000000000 +0100
@@ -13,7 +13,7 @@
 
 	$OUT .= "\n";
 	$OUT .= utf8("dn: uid=$key,$ldapBase\n");
-	$OUT .= utf8("objectClass: person\n");
+	$OUT .= utf8("objectClass: inetOrgPerson\n");
 	$OUT .= utf8("uid: $key\n");
 	$OUT .= utf8("cn: $name\n") if $name;
 	$OUT .= utf8("givenName: $first\n") if $first;
