diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory
--- e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory	2003-12-18 10:17:16.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/web/functions/directory	2007-01-13 10:52:02.000000000 -0700
@@ -95,7 +95,7 @@
     <page 
         name="First"  
         post-event="change_settings()" 
-        pre-event="turn_off_buttons()">
+        pre-event="print_status_message()">
         <description>DESCRIPTION</description>
 
         <field type="literal" value="get_ldap_base()">
@@ -148,8 +148,4 @@
         <subroutine src="print_button('SAVE')" />
     </page>
 
-    <page name="Done" pre-event="turn_off_buttons">
-        <description>SUCCESS</description>
-    </page>
-
 </form>
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/directory.pm mezzanine_patched_e-smith-ldap-4.12.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/directory.pm
--- e-smith-ldap-4.12.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/directory.pm	2005-04-14 19:24:58.000000000 -0600
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/directory.pm	2007-01-13 10:55:12.000000000 -0700
@@ -194,9 +194,9 @@
     #------------------------------------------------------------
 
     system ("/sbin/e-smith/signal-event ldap-update") == 0
-        or die ("Error occurred while updating system configuration.\n");
+        or return $fm->error('ERROR_UPDATING');
 
-    $fm->{cgi}->param( -name => 'wherenext', -value => 'Done' );
+    return $fm->success('SUCCESS');
 }
 
 # }}}
