--- e-smith-ldap-5.2.0/createlinks.ldap_update_later	2010-10-13 19:16:56.000000000 +0200
+++ e-smith-ldap-5.2.0/createlinks	2010-10-13 19:19:14.000000000 +0200
@@ -14,9 +14,9 @@
 	));
 }
 
-event_link("ldap-update", "group-create", "25");
+event_link("ldap-update", "group-create", "95");
 event_link("ldap-delete", "group-delete", "25");
-event_link("ldap-update", "user-create", "25");
+event_link("ldap-update", "user-create", "95");
 event_link("ldap-delete", "user-delete", "25");
 event_link("ldap-update", "user-lock", "25");
 
