diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory
--- e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory	2003-04-03 07:10:01.000000000 +1000
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/directory	2008-02-13 23:16:44.000000000 +1100
@@ -94,17 +94,7 @@
 <trans>Update with new defaults</trans>
 </entry>
     <entry>
-        <base>Configuration</base>
-        <trans>Configuration</trans>
-    </entry>
-
-    <entry>
         <base>Directory</base>
         <trans>Directory</trans>
     </entry>
-
-    <entry>
-        <base>SAVE</base>
-        <trans>Save</trans>
-    </entry>
 </lexicon>
