diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/createlinks mezzanine_patched_e-smith-ldap-4.12.0/createlinks
--- e-smith-ldap-4.12.0/createlinks	2006-03-15 14:17:43.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/createlinks	2007-02-20 12:23:25.000000000 -0700
@@ -32,4 +32,9 @@
 
 event_link("ldap-delete-dumps", "pre-restore", "25");
 
+foreach my $file (qw(/etc/ldap.conf /etc/ldap.secret /etc/cpu.conf))
+{
+    templates2events($file, qw(console-save bootstrap-console-save));
+}
+
 exit 0;
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/cpu.conf/all mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/cpu.conf/all
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/cpu.conf/all	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/cpu.conf/all	2007-02-20 12:31:48.000000000 -0700
@@ -0,0 +1,56 @@
+# See cpu.conf(5) for documentation
+
+[GLOBAL]
+DEFAULT_METHOD  = ldap
+CRACKLIB_DICTIONARY = /usr/lib/cracklib_dict
+
+[LDAP]
+LDAP_HOST       = 127.0.0.1
+LDAP_PORT       = 389
+# Can also use LDAP_URI = ldaps://localhost:389 for TLS support
+BIND_DN         = "cn=root,{ esmith::util::ldapBase ($DomainName); }"
+BIND_PASS       = { esmith::util::LdapPassword (); }
+USER_BASE       = ou=Users,{ esmith::util::ldapBase ($DomainName); }
+# replace account with inetOrgPerson if you want first or last name
+GROUP_BASE      = ou=Groups,{ esmith::util::ldapBase ($DomainName); }
+USER_OBJECT_CLASS       = account,posixAccount,shadowAccount,top
+GROUP_OBJECT_CLASS      = posixGroup,top
+USER_FILTER     = (objectClass=posixAccount)
+GROUP_FILTER    = (objectClass=posixGroup)
+USER_CN_STRING  = uid
+GROUP_CN_STRING = cn
+SKEL_DIR        = /etc/skel
+DEFAULT_SHELL   = /bin/bash
+HOME_DIRECTORY  = /home
+MAX_UIDNUMBER = 10000
+MIN_UIDNUMBER = 100
+MAX_GIDNUMBER = 10000
+MIN_GIDNUMBER = 101
+ID_MAX_PASSES = 1000
+USERGROUPS = yes
+USERS_GID = 100
+RANDOM = "false"
+PASSWORD_FILE = "/etc/passfile"
+SHADOW_FILE = "/etc/shadowfile"
+HASH = "sha1"
+#ADD_SCRIPT = "contrib/postaddscript.sh"
+#DEL_SCRIPT = "foo"
+SHADOWLASTCHANGE        = 11192
+SHADOWMAX               = 99999
+SHADOWWARING            = 7
+SHADOWEXPIRE            = -1
+SHADOWFLAG              = 134538308
+SHADOWMIN               = -1
+SHADOWINACTIVE          = -1
+
+[PASSWD]
+# Broken
+GROUP   =       1000
+HOME    =       /home
+INACTIVE =      -1
+#EXPIRE =
+SHELL   =       /bin/bash
+SKEL    =       /etc/skel
+COMMENT =       "Default Gecos"
+PASSWORD =      /etc/passwd
+SHADOW  =       /etc/shadow
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl	2007-02-20 13:47:40.000000000 -0700
@@ -0,0 +1 @@
+ssl no
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password	2007-02-20 13:47:52.000000000 -0700
@@ -0,0 +1 @@
+pam_password md5
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/30host mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/30host
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/30host	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/30host	2007-02-20 13:47:59.000000000 -0700
@@ -0,0 +1 @@
+host localhost
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40base mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40base
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40base	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40base	2007-02-20 13:48:27.000000000 -0700
@@ -0,0 +1,3 @@
+{
+    $OUT .= "base " . esmith::util::ldapBase ($DomainName);
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group	2007-02-20 13:48:40.000000000 -0700
@@ -0,0 +1,5 @@
+{
+    $OUT .= "nss_base_group ou=Groups,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd	2007-02-20 13:48:50.000000000 -0700
@@ -0,0 +1,5 @@
+{
+    $OUT .= "nss_base_passwd ou=Users,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn	2007-02-20 13:48:59.000000000 -0700
@@ -0,0 +1,3 @@
+{
+    $OUT .= "rootbinddn cn=root," . esmith::util::ldapBase ($DomainName);
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/50tls_cacertdir mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/50tls_cacertdir
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/50tls_cacertdir	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.conf/50tls_cacertdir	2007-02-20 13:49:06.000000000 -0700
@@ -0,0 +1 @@
+tls_cacertdir /etc/openldap/cacerts
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.secret mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.secret
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.secret	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/ldap.secret	2007-02-20 14:26:45.000000000 -0700
@@ -0,0 +1 @@
+{ esmith::util::LdapPassword (); }
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/10schema mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/10schema
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/10schema	2005-06-13 12:18:52.000000000 -0600
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/10schema	2007-02-20 14:27:56.000000000 -0700
@@ -5,3 +5,4 @@
 include         /etc/openldap/schema/nis.schema
 include         /etc/openldap/schema/redhat/rfc822-MailMember.schema
 include         /etc/openldap/schema/redhat/autofs.schema
+include         /etc/openldap/schema/samba.schema
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/20schemacheck mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/20schemacheck
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/20schemacheck	2002-03-25 12:17:42.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/20schemacheck	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-schemacheck	off
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/95acls mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/95acls
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/95acls	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/etc/openldap/slapd.conf/95acls	2007-02-20 14:30:10.000000000 -0700
@@ -0,0 +1,12 @@
+# database access control definitions
+ access to attr=userPassword
+ by self write
+ by anonymous auth
+ by dn="cn=root,{ esmith::util::ldapBase ($DomainName); }" write
+ by * none
+
+ access to *
+ by self write
+ by dn="cn=root,{ esmith::util::ldapBase ($DomainName); }" write
+ by * read
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10domain mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10domain
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10domain	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10domain	2007-02-20 14:32:12.000000000 -0700
@@ -0,0 +1,7 @@
+{
+    my ($dc) = split(/\./, $DomainName);
+    $OUT .= "dn: $ldapBase\n";
+    $OUT .= "objectClass : top\n";
+    $OUT .= "objectClass : domain\n";
+    $OUT .= "dc : $dc\n";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	2005-03-23 10:36:25.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/10organisation	1969-12-31 17:00:00.000000000 -0700
@@ -1,4 +0,0 @@
-{
-    $OUT .= "dn: $ldapBase\n";
-    $OUT .= "objectClass : organization\n";
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/12UserGroupContainers mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/12UserGroupContainers
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/12UserGroupContainers	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/12UserGroupContainers	2007-02-20 14:32:32.000000000 -0700
@@ -0,0 +1,9 @@
+{
+    foreach my $ou (qw(Users Groups UsersPasscodes))
+    {
+       $OUT .= "dn: ou=$ou,$ldapBase\n";
+       $OUT .= "objectClass : top\n";
+       $OUT .= "objectClass : organizationalUnit\n";
+       $OUT .= "ou : $ou\n\n";
+    }
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2006-03-15 14:17:43.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2007-02-20 14:34:44.000000000 -0700
@@ -1,13 +1,48 @@
 {
+    my $next_gid = $MinUid || 5000;
+    foreach my $key (qw(shared admin root www))
+    {
+        my $gid = getgrnam($key);
+        $OUT .= "\n";
+        $OUT .= "dn: cn=$key,ou=Groups,$ldapBase\n";
+        $OUT .= "objectClass: posixGroup\n";
+        $OUT .= "gidNumber: $gid\n";
+        $OUT .= "cn: $key\n";
+    }
     foreach ($a->groups)
     {
-	my $key = $_->key;
-	my $desc = $_->prop('Description');
+        my $key = $_->key;
+        my $desc = $_->prop('Description');
+        my $members = $_->prop('Members');
+        my $uid = $_->prop('Uid') || getpwnam($key) || $next_gid;
+        my $gid = $_->prop('Gid') || getgrnam($key) || $next_gid;
+        $next_gid = $gid + 1 unless ($next_gid > $gid);
 
-	$OUT .= "\n";
-	$OUT .= "dn: uid=$key,$ldapBase\n";
-	$OUT .= "objectClass: group\n";
-	$OUT .= "mail: $key\@$DomainName\n";
-	$OUT .= utf8("cn: $desc\n") if $desc;
+        $OUT .= "\n";
+        $OUT .= "dn: cn=$key,ou=Groups,$ldapBase\n";
+        $OUT .= "objectClass: posixGroup\n";
+        $OUT .= "objectClass: top\n";
+        $OUT .= "structuralObjectClass: posixGroup\n";
+        $OUT .= "cn: $key\n";
+        $OUT .= "gidNumber: $gid\n";
+        foreach (split(/,/, $members))
+        {
+            $OUT .= "memberUid: $_\n";
+        }
+        $OUT .= utf8("description: $desc\n") if $desc;
+        $OUT .= "\n";
+        $OUT .= "dn: uid=$key,$ldapBase\n";
+        $OUT .= "cn: $key\n";
+        $OUT .= "objectClass: account\n";
+        $OUT .= "objectClass: posixAccount\n";
+        $OUT .= "objectClass: shadowAccount\n";
+        $OUT .= "objectClass: top\n";
+        $OUT .= "structuralObjectClass: account\n";
+        $OUT .= "gidNumber: $gid\n";
+        $OUT .= "uidNumber: $uid\n";
+        $OUT .= "uid: $key\n";
+        $OUT .= "homeDirectory: /home/e-smith\n";
+        $OUT .= "loginShell: /bin/false\n";
+        $OUT .= utf8("gecos: $desc\n") if $desc;
     }
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2005-04-15 13:30:38.000000000 -0600
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50users	2007-02-20 14:35:50.000000000 -0700
@@ -1,28 +1,50 @@
 {
+    foreach my $key (qw(admin root www))
+    {
+        $OUT .= "\n";
+        $OUT .= "dn: uid=$key,ou=Users,$ldapBase\n";
+        $OUT .= "objectClass: account\n";
+        $OUT .= "uid: $key\n";
+    }
+
     foreach my $user ($a->users)
     {
-	my $key = $user->key;
-	my $first = $user->prop('FirstName') || '';
-	my $last = $user->prop('LastName') || '';
-	my $name = "$first $last";
-	my $phone = $user->prop('Phone');
-	my $company = $user->prop('Company');
-	my $dept = $user->prop('Dept');
-	my $city = $user->prop('City');
-	my $street = $user->prop('Street');
+        my $key = $user->key;
+
+        my $first = $user->prop('FirstName') || '';
+        my $last = $user->prop('LastName') || '';
+        my $full_name = "$first $last";
+        my $phone = $user->prop('Phone');
+        my $company = $user->prop('Company');
+        my $dept = $user->prop('Dept');
+        my $city = $user->prop('City');
+        my $street = $user->prop('Street');
+        my $uid = $user->prop('Uid') || getpwnam($key);
+        my $shell = $user->prop('Shell') || '/usr/bin/rssh';
 
-	$OUT .= "\n";
-	$OUT .= utf8("dn: uid=$key,$ldapBase\n");
-	$OUT .= utf8("objectClass: person\n");
-	$OUT .= utf8("uid: $key\n");
-	$OUT .= utf8("cn: $name\n") if $name;
-	$OUT .= utf8("givenName: $first\n") if $first;
-	$OUT .= utf8("sn: $last\n") if $last;
-	$OUT .= utf8("mail: $key\@$DomainName\n");
-	$OUT .= utf8("telephoneNumber: $phone\n") if $phone;
-	$OUT .= utf8("o: $company\n") if $company;
-	$OUT .= utf8("ou: $dept\n") if $dept;
-	$OUT .= utf8("l: $city\n") if $city;
-	$OUT .= utf8("street: $street\n") if $street;
+        $OUT .= "\n";
+        $OUT .= utf8("dn: uid=$key,ou=Users,$ldapBase\n");
+        $OUT .= utf8("objectClass: posixAccount\n");
+        $OUT .= utf8("objectClass: inetOrgPerson\n");
+        $OUT .= utf8("uid: $key\n");
+        $OUT .= utf8("uidNumber: $uid\n");
+        $OUT .= utf8("gidNumber: $uid\n");
+        $OUT .= utf8("cn: $key\n");
+        $OUT .= utf8("loginShell: $shell\n");
+        $OUT .= utf8("homeDirectory: /home/e-smith/files/users/$key\n");
+        $OUT .= utf8("gecos: $full_name\n") if $full_name;
+        $OUT .= utf8("givenName: $first\n") if $first;
+        $OUT .= utf8("sn: $last\n") if $last;
+        $OUT .= utf8("mail: $key\@$DomainName\n");
+        $OUT .= utf8("telephoneNumber: $phone\n") if $phone;
+        $OUT .= utf8("o: $company\n") if $company;
+        $OUT .= utf8("departmentNumber: $dept\n") if $dept;
+        $OUT .= utf8("l: $city\n") if $city;
+        $OUT .= utf8("street: $street\n") if $street;
+        $OUT .= "\n";
+        $OUT .= utf8("dn: cn=$key,$ldapBase\n");
+        $OUT .= utf8("objectClass: posixGroup\n");
+        $OUT .= utf8("gidNumber: $uid\n");
+        $OUT .= utf8("cn: $key\n");
     }
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/cpu.conf mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/cpu.conf
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/cpu.conf	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/cpu.conf	2007-02-20 14:36:51.000000000 -0700
@@ -0,0 +1 @@
+PERMS=0600
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/ldap.secret mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/ldap.secret
--- e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/ldap.secret	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/e-smith/templates.metadata/etc/ldap.secret	2007-02-20 14:37:12.000000000 -0700
@@ -0,0 +1 @@
+PERMS=0600
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/etc/openldap/schema/samba.schema mezzanine_patched_e-smith-ldap-4.12.0/root/etc/openldap/schema/samba.schema
--- e-smith-ldap-4.12.0/root/etc/openldap/schema/samba.schema	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/etc/openldap/schema/samba.schema	2004-09-08 14:52:24.000000000 -0600
@@ -0,0 +1,480 @@
+##
+## schema file for OpenLDAP 2.x
+## Schema for storing Samba user accounts and group maps in LDAP
+## OIDs are owned by the Samba Team
+##
+## Prerequisite schemas - uid         (cosine.schema)
+##                      - displayName (inetorgperson.schema)
+##                      - gidNumber   (nis.schema)
+##
+## 1.3.6.1.4.1.7165.2.1.x - attributetypes
+## 1.3.6.1.4.1.7165.2.2.x - objectclasses
+##
+## Printer support
+## 1.3.6.1.4.1.7165.2.3.1.x - attributetypes
+## 1.3.6.1.4.1.7165.2.3.2.x - objectclasses
+##
+## ----- READ THIS WHEN ADDING A NEW ATTRIBUTE OR OBJECT CLASS ------
+##
+## Run the 'get_next_oid' bash script in this directory to find the 
+## next available OID for attribute type and object classes.
+##
+##   $ ./get_next_oid
+##   attributetype ( 1.3.6.1.4.1.7165.2.1.XX NAME ....
+##   objectclass ( 1.3.6.1.4.1.7165.2.2.XX NAME ....
+##
+## Also ensure that new entries adhere to the declaration style
+## used throughout this file
+##
+##    <attributetype|objectclass> ( 1.3.6.1.4.1.7165.2.XX.XX NAME ....
+##                               ^ ^                        ^
+##
+## The spaces are required for the get_next_oid script (and for 
+## readability).
+##
+## ------------------------------------------------------------------
+
+# objectIdentifier SambaRoot 1.3.6.1.4.1.7165
+# objectIdentifier Samba3 SambaRoot:2
+# objectIdentifier Samba3Attrib Samba3:1
+# objectIdentifier Samba3ObjectClass Samba3:2
+
+########################################################################
+##                            HISTORICAL                              ##
+########################################################################
+
+##
+## Password hashes
+##
+#attributetype ( 1.3.6.1.4.1.7165.2.1.1 NAME 'lmPassword'
+#	DESC 'LanManager Passwd'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{32} SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.2 NAME 'ntPassword'
+#	DESC 'NT Passwd'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{32} SINGLE-VALUE )
+
+##
+## Account flags in string format ([UWDX     ])
+##
+#attributetype ( 1.3.6.1.4.1.7165.2.1.4 NAME 'acctFlags'
+#	DESC 'Account Flags'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{16} SINGLE-VALUE )
+
+##
+## Password timestamps & policies
+##
+#attributetype ( 1.3.6.1.4.1.7165.2.1.3 NAME 'pwdLastSet'
+#	DESC 'NT pwdLastSet'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.5 NAME 'logonTime'
+#	DESC 'NT logonTime'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.6 NAME 'logoffTime'
+#	DESC 'NT logoffTime'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.7 NAME 'kickoffTime'
+#	DESC 'NT kickoffTime'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.8 NAME 'pwdCanChange'
+#	DESC 'NT pwdCanChange'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.9 NAME 'pwdMustChange'
+#	DESC 'NT pwdMustChange'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+##
+## string settings
+##
+#attributetype ( 1.3.6.1.4.1.7165.2.1.10 NAME 'homeDrive'
+#	DESC 'NT homeDrive'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{4} SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.11 NAME 'scriptPath'
+#	DESC 'NT scriptPath'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{255} SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.12 NAME 'profilePath'
+#	DESC 'NT profilePath'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{255} SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.13 NAME 'userWorkstations'
+#	DESC 'userWorkstations'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{255} SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.17 NAME 'smbHome'
+#	DESC 'smbHome'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{128} )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.18 NAME 'domain'
+#	DESC 'Windows NT domain to which the user belongs'
+#	EQUALITY caseIgnoreIA5Match
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{128} )
+
+##
+## user and group RID
+##
+#attributetype ( 1.3.6.1.4.1.7165.2.1.14 NAME 'rid'
+#	DESC 'NT rid'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+#attributetype ( 1.3.6.1.4.1.7165.2.1.15 NAME 'primaryGroupID'
+#	DESC 'NT Group RID'
+#	EQUALITY integerMatch
+#	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+##
+## The smbPasswordEntry objectclass has been depreciated in favor of the
+## sambaAccount objectclass
+##
+#objectclass ( 1.3.6.1.4.1.7165.2.2.1 NAME 'smbPasswordEntry' SUP top AUXILIARY
+#        DESC 'Samba smbpasswd entry'
+#        MUST ( uid $ uidNumber )
+#        MAY  ( lmPassword $ ntPassword $ pwdLastSet $ acctFlags ))
+
+#objectclass ( 1.3.6.1.4.1.7165.2.2.2 NAME 'sambaAccount' SUP top STRUCTURAL
+#	DESC 'Samba Account'
+#	MUST ( uid $ rid )
+#	MAY  ( cn $ lmPassword $ ntPassword $ pwdLastSet $ logonTime $
+#               logoffTime $ kickoffTime $ pwdCanChange $ pwdMustChange $ acctFlags $
+#               displayName $ smbHome $ homeDrive $ scriptPath $ profilePath $
+#               description $ userWorkstations $ primaryGroupID $ domain ))
+
+#objectclass ( 1.3.6.1.4.1.7165.2.2.3 NAME 'sambaAccount' SUP top AUXILIARY
+#	DESC 'Samba Auxiliary Account'
+#	MUST ( uid $ rid )
+#	MAY  ( cn $ lmPassword $ ntPassword $ pwdLastSet $ logonTime $
+#              logoffTime $ kickoffTime $ pwdCanChange $ pwdMustChange $ acctFlags $
+#              displayName $ smbHome $ homeDrive $ scriptPath $ profilePath $
+#              description $ userWorkstations $ primaryGroupID $ domain ))
+
+########################################################################
+##                        END OF HISTORICAL                           ##
+########################################################################
+
+#######################################################################
+##                Attributes used by Samba 3.0 schema                ##
+#######################################################################
+
+##
+## Password hashes
+##
+attributetype ( 1.3.6.1.4.1.7165.2.1.24 NAME 'sambaLMPassword'
+	DESC 'LanManager Password'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{32} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.25 NAME 'sambaNTPassword'
+	DESC 'MD4 hash of the unicode password'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{32} SINGLE-VALUE )
+
+##
+## Account flags in string format ([UWDX     ])
+##
+attributetype ( 1.3.6.1.4.1.7165.2.1.26 NAME 'sambaAcctFlags'
+	DESC 'Account Flags'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{16} SINGLE-VALUE )
+
+##
+## Password timestamps & policies
+##
+attributetype ( 1.3.6.1.4.1.7165.2.1.27 NAME 'sambaPwdLastSet'
+	DESC 'Timestamp of the last password update'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.28 NAME 'sambaPwdCanChange'
+	DESC 'Timestamp of when the user is allowed to update the password'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.29 NAME 'sambaPwdMustChange'
+	DESC 'Timestamp of when the password will expire'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.30 NAME 'sambaLogonTime'
+	DESC 'Timestamp of last logon'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.31 NAME 'sambaLogoffTime'
+	DESC 'Timestamp of last logoff'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.32 NAME 'sambaKickoffTime'
+	DESC 'Timestamp of when the user will be logged off automatically'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.48 NAME 'sambaBadPasswordCount'
+	DESC 'Bad password attempt count'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.49 NAME 'sambaBadPasswordTime'
+	DESC 'Time of the last bad password attempt'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.55 NAME 'sambaLogonHours'
+	DESC 'Logon Hours'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{42} SINGLE-VALUE )
+
+##
+## string settings
+##
+attributetype ( 1.3.6.1.4.1.7165.2.1.33 NAME 'sambaHomeDrive'
+	DESC 'Driver letter of home directory mapping'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{4} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.34 NAME 'sambaLogonScript'
+	DESC 'Logon script path'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{255} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.35 NAME 'sambaProfilePath'
+	DESC 'Roaming profile path'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{255} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.36 NAME 'sambaUserWorkstations'
+	DESC 'List of user workstations the user is allowed to logon to'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{255} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.37 NAME 'sambaHomePath'
+	DESC 'Home directory UNC path'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{128} )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.38 NAME 'sambaDomainName'
+	DESC 'Windows NT domain to which the user belongs'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{128} )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.47 NAME 'sambaMungedDial'
+	DESC ''
+	EQUALITY caseExactMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{1050} )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.54 NAME 'sambaPasswordHistory'
+	DESC 'Concatenated MD4 hashes of the unicode passwords used on this account'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{32} )
+
+##
+## SID, of any type
+##
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.20 NAME 'sambaSID'
+	DESC 'Security ID'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{64} SINGLE-VALUE )
+
+
+##
+## Primary group SID, compatible with ntSid
+##
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.23 NAME 'sambaPrimaryGroupSID'
+	DESC 'Primary Group Security ID'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{64} SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.51 NAME 'sambaSIDList'
+	DESC 'Security ID List'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{64} )
+
+##
+## group mapping attributes
+##
+attributetype ( 1.3.6.1.4.1.7165.2.1.19 NAME 'sambaGroupType'
+	DESC 'NT Group Type'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+##
+## Store info on the domain
+##
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.21 NAME 'sambaNextUserRid'
+	DESC 'Next NT rid to give our for users'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.22 NAME 'sambaNextGroupRid'
+	DESC 'Next NT rid to give out for groups'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.39 NAME 'sambaNextRid'
+	DESC 'Next NT rid to give out for anything'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.40 NAME 'sambaAlgorithmicRidBase'
+	DESC 'Base at which the samba RID generation algorithm should operate'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.41 NAME 'sambaShareName'
+	DESC 'Share Name'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.42 NAME 'sambaOptionName'
+	DESC 'Option Name'
+	EQUALITY caseIgnoreMatch
+	SUBSTR caseIgnoreSubstringsMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{256} )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.43 NAME 'sambaBoolOption'
+	DESC 'A boolean option'
+	EQUALITY booleanMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.7 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.44 NAME 'sambaIntegerOption'
+	DESC 'An integer option'
+	EQUALITY integerMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.45 NAME 'sambaStringOption'
+	DESC 'A string option'
+	EQUALITY caseExactIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26 SINGLE-VALUE )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.46 NAME 'sambaStringListOption'
+	DESC 'A string list option'
+	EQUALITY caseIgnoreMatch
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 )
+
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.50 NAME 'sambaPrivName' 
+	SUP name )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.52 NAME 'sambaPrivilegeList'
+	DESC 'Privileges List'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{64} )
+
+attributetype ( 1.3.6.1.4.1.7165.2.1.53 NAME 'sambaTrustFlags'
+	DESC 'Trust Password Flags'
+	EQUALITY caseIgnoreIA5Match
+	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26 )
+
+
+#######################################################################
+##              objectClasses used by Samba 3.0 schema               ##
+#######################################################################
+
+## The X.500 data model (and therefore LDAPv3) says that each entry can
+## only have one structural objectclass.  OpenLDAP 2.0 does not enforce
+## this currently but will in v2.1
+
+##
+## added new objectclass (and OID) for 3.0 to help us deal with backwards
+## compatibility with 2.2 installations (e.g. ldapsam_compat)  --jerry
+##
+objectclass ( 1.3.6.1.4.1.7165.2.2.6 NAME 'sambaSamAccount' SUP top AUXILIARY
+	DESC 'Samba 3.0 Auxilary SAM Account'
+	MUST ( uid $ sambaSID )
+	MAY  ( cn $ sambaLMPassword $ sambaNTPassword $ sambaPwdLastSet $
+	       sambaLogonTime $ sambaLogoffTime $ sambaKickoffTime $
+	       sambaPwdCanChange $ sambaPwdMustChange $ sambaAcctFlags $
+               displayName $ sambaHomePath $ sambaHomeDrive $ sambaLogonScript $
+	       sambaProfilePath $ description $ sambaUserWorkstations $
+	       sambaPrimaryGroupSID $ sambaDomainName $ sambaMungedDial $
+	       sambaBadPasswordCount $ sambaBadPasswordTime $
+	       sambaPasswordHistory $ sambaLogonHours))
+
+##
+## Group mapping info
+##
+objectclass ( 1.3.6.1.4.1.7165.2.2.4 NAME 'sambaGroupMapping' SUP top AUXILIARY
+	DESC 'Samba Group Mapping'
+	MUST ( gidNumber $ sambaSID $ sambaGroupType )
+	MAY  ( displayName $ description $ sambaSIDList ))
+
+##
+## Trust password for trust relationships (any kind)
+##
+objectclass ( 1.3.6.1.4.1.7165.2.2.14 NAME 'sambaTrustPassword' SUP top STRUCTURAL
+	DESC 'Samba Trust Password'
+	MUST ( sambaDomainName $ sambaNTPassword $ sambaTrustFlags )
+	MAY ( sambaSID $ sambaPwdLastSet ))
+
+##
+## Whole-of-domain info
+##
+objectclass ( 1.3.6.1.4.1.7165.2.2.5 NAME 'sambaDomain' SUP top STRUCTURAL
+	DESC 'Samba Domain Information'
+	MUST ( sambaDomainName $ 
+	       sambaSID ) 
+	MAY ( sambaNextRid $ sambaNextGroupRid $ sambaNextUserRid $
+	      sambaAlgorithmicRidBase ) )
+
+##
+## used for idmap_ldap module
+##
+objectclass ( 1.3.6.1.4.1.7165.2.2.7 NAME 'sambaUnixIdPool' SUP top AUXILIARY
+        DESC 'Pool for allocating UNIX uids/gids'
+        MUST ( uidNumber $ gidNumber ) )
+
+
+objectclass ( 1.3.6.1.4.1.7165.2.2.8 NAME 'sambaIdmapEntry' SUP top AUXILIARY
+        DESC 'Mapping from a SID to an ID'
+        MUST ( sambaSID )
+	MAY ( uidNumber $ gidNumber ) )
+
+objectclass ( 1.3.6.1.4.1.7165.2.2.9 NAME 'sambaSidEntry' SUP top STRUCTURAL
+	DESC 'Structural Class for a SID'
+	MUST ( sambaSID ) )
+
+objectclass ( 1.3.6.1.4.1.7165.1.2.2.10 NAME 'sambaConfig' SUP top AUXILIARY
+	DESC 'Samba Configuration Section'
+	MAY ( description ) )
+
+objectclass ( 1.3.6.1.4.1.7165.2.2.11 NAME 'sambaShare' SUP top STRUCTURAL
+	DESC 'Samba Share Section'
+	MUST ( sambaShareName )
+	MAY ( description ) )
+
+objectclass ( 1.3.6.1.4.1.7165.2.2.12 NAME 'sambaConfigOption' SUP top STRUCTURAL
+	DESC 'Samba Configuration Option'
+	MUST ( sambaOptionName )
+	MAY ( sambaBoolOption $ sambaIntegerOption $ sambaStringOption $ 
+	      sambaStringListoption $ description ) )
+
+
+objectclass ( 1.3.6.1.4.1.7165.2.2.13 NAME 'sambaPrivilege' SUP top AUXILIARY
+	DESC 'Samba Privilege'
+	MUST ( sambaSID )
+	MAY ( sambaPrivilegeList ) )
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.12.0/root/var/service/ldap/run mezzanine_patched_e-smith-ldap-4.12.0/root/var/service/ldap/run
--- e-smith-ldap-4.12.0/root/var/service/ldap/run	2007-02-20 14:43:56.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.12.0/root/var/service/ldap/run	2007-02-20 14:42:54.000000000 -0700
@@ -1,29 +1,41 @@
 #! /bin/sh
 
 domain=$(/sbin/e-smith/config get DomainName)
+old_domain=$(slapcat 2>/dev/null | sed -n -e 's/dn: dc=//' -e 's/,dc=/./gp' -e 2q)
+
 ldif="/home/e-smith/db/ldap/$domain.ldif"
+old_ldif="/home/e-smith/db/ldap/$old_domain.ldif"
 
-if [ -e ldif ]
+if [ "$old_domain" != "$domain" ]
 then
-    old_ldif=$(readlink ldif)
-    if [ "$old_ldif" != "$ldif" ]
-    then
-	# The domain name has changed, so we need to delete
-	# the old directory contents. We still have the old
-	# dump.
-	find /var/lib/ldap -type f | xargs rm -f
-    fi
+    # The domain name has changed, so we need to delete
+    # the old directory contents. We still have the old
+    # dump.
+    find /var/lib/ldap -type f | xargs rm -f
 fi
 
 # Set up symlink for ldap dump at shutdown
 ln -sf $ldif ./ldif
 
 # Prime directory if required
-if [ \! -f /var/lib/ldap/nextid.dbb -a -f $ldif ]
+if [ \! -f /var/lib/ldap/nextid.dbb ]
 then
-    sed 's/objectClass: group/objectClass: posixGroup/' < $ldif | \
-      setuidgid ldap slapadd -c
+    if [ -e "$old_ldif" ]
+    then
+       old_base_dn=$(echo $old_domain | sed -e 's/\./,dc=/g' -e 's/^/dc=/')
+       base_dn=$(echo $domain | sed -e 's/\./,dc=/g' -e 's/^/dc=/')
+       sed -e "s/$old_base_dn/$base_dn/" \
+           -e 's/objectClass: group/objectClass: posixGroup/' < $old_ldif | \
+           setuidgid ldap slapadd -c
+    else
+       if [ \! -e "$ldif" ]
+       then
+           /sbin/e-smith/expand-template /home/e-smith/db/ldap/ldif
+       fi
+       sed -e 's/objectClass: group/objectClass: posixGroup/' < $ldif | \
+           setuidgid ldap slapadd -c
+    fi
 fi
 
 # Now run daemon
-exec /usr/sbin/slapd -4 -u ldap -d 0
+exec /usr/sbin/slapd -u ldap -d 0
