diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.11.3/root/etc/e-smith/events/actions/ldap-update mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/events/actions/ldap-update
--- e-smith-ldap-4.11.3/root/etc/e-smith/events/actions/ldap-update	2006-01-22 21:02:52.000000000 -0500
+++ mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/events/actions/ldap-update	2006-01-22 12:03:13.000000000 -0500
@@ -100,14 +100,20 @@
     if ($type eq 'user')
     {
 	my $name = $acct->prop('FirstName') . " " . $acct->prop('LastName');
+	utf8::upgrade($name);
 	my $first = $acct->prop('FirstName') || '';
+	utf8::upgrade($first);
 	my $last = $acct->prop('LastName') || '';
+	utf8::upgrade($last);
 	my $phone = $acct->prop('Phone') || '';
 	my $company = $acct->prop('Company') || '';
+	utf8::upgrade($company);
 	my $dept = $acct->prop('Dept') || '';
+	utf8::upgrade($dept);
 	my $city = $acct->prop('City') || '';
+	utf8::upgrade($city);
 	my $street = $acct->prop('Street') || '';
-
+	utf8::upgrade($street);
 	push @attrs, (objectClass => 'person');
 	push @attrs, (uid => $key);
 
@@ -128,7 +134,7 @@
 
 	my $key = $acct->key;
 	my $desc = $acct->prop('Description') || '';
-
+	utf8::upgrade($desc);
 	push @attrs, (cn => $desc) unless $desc =~ /^\s*$/;
 	push @attrs, (mail => "$key\@$domain");
 	push @attrs, (telephoneNumber => $phone) unless $phone =~ /^\s*$/;
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups
--- e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2005-04-15 15:29:58.000000000 -0400
+++ mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/50groups	2006-01-22 12:50:04.000000000 -0500
@@ -8,6 +8,6 @@
 	$OUT .= "dn: uid=$key,$ldapBase\n";
 	$OUT .= "objectClass: group\n";
 	$OUT .= "mail: $key\@$DomainName\n";
-	$OUT .= "cn: $desc\n" if $desc;
+	$OUT .= utf8("cn: $desc\n") if $desc;
     }
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin
--- e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin	2005-04-01 16:47:02.000000000 -0500
+++ mezzanine_patched_e-smith-ldap-4.11.3/root/etc/e-smith/templates/home/e-smith/db/ldap/ldif/template-begin	2006-01-22 21:02:33.000000000 -0500
@@ -1,14 +1,14 @@
 {
     use esmith::AccountsDB;
     use esmith::util;
-    #use Unicode::String;
 
     $a = esmith::AccountsDB->open_ro;
     $ldapBase = esmith::util::ldapBase ($DomainName);
     sub utf8
     {
-	#Unicode::String::latin1(shift);
-	return shift;
+	my $t = shift;
+	utf8::encode($t);
+	return $t;
     }
     $OUT = "";
 }
