diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.11.1/root/etc/e-smith/db/configuration/defaults/ldap/TCPPort mezzanine_patched_e-smith-ldap-4.11.1/root/etc/e-smith/db/configuration/defaults/ldap/TCPPort
--- e-smith-ldap-4.11.1/root/etc/e-smith/db/configuration/defaults/ldap/TCPPort	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-ldap-4.11.1/root/etc/e-smith/db/configuration/defaults/ldap/TCPPort	2005-07-20 19:03:52.967161272 -0600
@@ -0,0 +1 @@
+389
diff -Nur -x '*.orig' -x '*.rej' e-smith-ldap-4.11.1/root/etc/e-smith/templates/etc/rc.d/init.d/masq/45AllowLDAP mezzanine_patched_e-smith-ldap-4.11.1/root/etc/e-smith/templates/etc/rc.d/init.d/masq/45AllowLDAP
--- e-smith-ldap-4.11.1/root/etc/e-smith/templates/etc/rc.d/init.d/masq/45AllowLDAP	2002-08-15 14:47:17.000000000 -0600
+++ mezzanine_patched_e-smith-ldap-4.11.1/root/etc/e-smith/templates/etc/rc.d/init.d/masq/45AllowLDAP	1969-12-31 17:00:00.000000000 -0700
@@ -1,6 +0,0 @@
-{
-    my $status = $ldap{status} || 'disabled';
-    my $access = $ldap{access} || 'private';
-
-    $OUT = allow_tcp_in(389, ($status eq "enabled" && $access eq 'public'));
-}
